### PR TITLE
Added install info to README for Gradle users.

### PR DESCRIPTION
**Pull request by kkovach**

### CODE REVIEW

**Comment by kkovach** [comment]
Signed the developer agreement.

File: README.md [hunk 1]
```diff
@@ -79,6 +80,39 @@ then add this to your maven configuration:
 </dependencies>
 ```
 
+### Gradle
+For Gradle there are 2 plugins that help you keep from unnecessarily compiling the `dagger-compiler` library into your project. One for POJO projects and another for Android projects. They allow you to add the `dagger-compiler` artifact as an `apt` dependency.
+```groovy
+buildscript {
+  repositories {
+    // For POJO projects and the net.ltgt.gradle:gradle-apt-plugin
```
Comment (JakeWharton) [inline README.md]: POJO projects? I believe you're looking for "Java projects" here.

File: README.md [hunk 2]
```diff
@@ -79,6 +80,39 @@ then add this to your maven configuration:
...
+// Apply plugin for POJO projects
+apply plugin: 'net.ltgt.apt'
+
+// Apply plugin for Android projects
+apply plugin: 'com.neenbedankt.android-apt'
+
+// Add Dagger dependencies (independent of project type)
+dependencies {
+  compile group: 'com.google.dagger', name: 'dagger', version: '2.0.2'
+  apt group: 'com.google.dagger', name: 'dagger-compiler', version: '2.0.2'
```
Comment (JakeWharton) [inline README.md]: Use the collapsed syntax:
```groovy
'com.google.dagger:dagger:2.0.2'
```
etc. Don't waste time with a verbose map.

File: README.md [hunk 3]
```diff
@@ -79,6 +80,39 @@ then add this to your maven configuration:
...
+    // For POJO projects and the net.ltgt.gradle:gradle-apt-plugin
+    maven {
+      url "https://plugins.gradle.org/m2/"
+    }
+    // For Android projects and the com.neenbedankt.gradle.plugins:android-apt plugin
+    mavenCentral()
+  }
+  dependencies {
+    // For POJO projects
+    classpath "net.ltgt.gradle:gradle-apt-plugin:0.5"
```
Comment (JakeWharton) [inline README.md]: Why not use the shorter `plugin` block syntax that combines the repository, dependency, and appy lines into a single line?

File: README.md [hunk 4]
```diff
@@ -79,6 +80,39 @@ then add this to your maven configuration:
 </dependencies>
 ```
 
+### Gradle
+For Gradle there are 2 plugins that help you keep from unnecessarily compiling the `dagger-compiler` library into your project. One for POJO projects and another for Android projects. They allow you to add the `dagger-compiler` artifact as an `apt` dependency.
```
Comment (JakeWharton) [inline README.md]: I would split this into two blocks. One for Java Gradle and one for Android Gradle. You'll only confuse people by mixing them.

File: README.md [hunk 5]
```diff
@@ -79,6 +80,44 @@ then add this to your maven configuration:
...
 
+### Java Gradle
+```groovy
+// Add plugin to classpath
+plugins {
+  id "net.ltgt.apt" version "0.5"
+}
+
+// Apply plugin
+apply plugin: 'net.ltgt.apt'
```
Comment (tbroyer) [inline README.md]: You don't need that line with the `plugins` syntax.

FWIW, I like to link to the plugin portal for instructions: https://plugins.gradle.org/plugin/net.ltgt.apt

File: README.md [hunk 6]
```diff
@@ -79,6 +80,41 @@ then add this to your maven configuration:
...
+// Add Dagger dependencies
+dependencies {
+  compile 'com.google.dagger:dagger:2.0.2'
+  apt 'com.google.dagger:dagger-compiler:2.0.2'
+}
+```
+
+### Android Gradle
+```groovy
+// Add plugin to classpath
```
Comment (gk5885) [inline README.md]: the android apt plugin should get a link as well

**Comment by gk5885** [comment]
This is fantastic.  Thanks.  One small change and we should be good.

**Comment by ronshapiro** [comment]
@kkovach can you rebase against master and squash your commits? Then we can pull this in. Thanks for doing this!

**Comment by kkovach** [comment]
Hell. Deleted the branch in my fork. Resubmitting pr now.

**Comment by NightlyNexus** [comment]
@kkovach Assuming you did rebase, Git was probably rightly complaining that you were trying to push to a different history. When you squashed, you rewrote your branch's history. You can force the overwrite with a `push --force`.

**Pull request closed without merging**